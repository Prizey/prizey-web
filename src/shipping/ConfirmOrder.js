import React, { Fragment } from 'react'
import { connect } from 'react-redux'
import { New } from 'croods'
import { withStyles } from '@material-ui/core/styles'
import { Button, Typography, CircularProgress } from '@material-ui/core'
import { Redirect } from '@reach/router'

import Layout from 'design/Layout/Layout'
import ErrorComponent from 'design/Error/Error'
import PlayAgain from 'design/PlayAgain'

import { fieldList } from 'auth/UserForm'

const styles = theme => ({
  field: {
    marginTop: theme.spacing.sm,
  },
  title: {
    marginTop: parseInt(theme.spacing.lg, 10) * -0.5,
  },
  value: {
    opacity: 0.8,
  },
})

export const handleCancel = navigate => () => navigate('/')
export const handleShipIt = (product = {}, create) => () =>
  create({ productId: product.id || 0 })

export const handleAfterCreate = navigate => () =>
  navigate('/shipping-confirmation')

export const mapState = ({ basket = {} }) => ({
  product: basket.product,
})

export class ConfirmOrderComponent extends React.Component {
  state = {
    dialogIsOpen: false,
  }

  renderHeader = ({ classes, navigate }) => (
    <Fragment>
      <PlayAgain
        aria-label="Play again modal"
        isOpen={this.state.dialogIsOpen}
        close={() => this.setState({ dialogIsOpen: false })}
        confirm={handleCancel(navigate)}
      >
        This action can&apos;t be undone.
      </PlayAgain>

      <Typography align="center" variant="h5" className={classes.title}>
        Confirm Order
      </Typography>
    </Fragment>
  )

  renderFields = ({ classes, currentUser }) =>
    fieldList.map(field => (
      <Typography key={field.name} className={classes.field} variant="body2">
        {field.label}:{' '}
        <span className={classes.value}>{currentUser[field.name]}</span>
      </Typography>
    ))

  renderBottom = ({ product, create, creating, error }) => (
    <Fragment>
      <div style={{ textAlign: 'center' }}>
        {creating && <CircularProgress color="primary" size={36} />}
        {error && <ErrorComponent>{error}</ErrorComponent>}
      </div>

      <Button
        variant="contained"
        color="primary"
        fullWidth
        onClick={handleShipIt(product, create)}
      >
        SHIP IT!
      </Button>
      <Button
        aria-label="Cancel"
        variant="outlined"
        color="primary"
        fullWidth
        onClick={() => this.setState({ dialogIsOpen: true })}
      >
        CANCEL
      </Button>
    </Fragment>
  )

  render() {
    const { product, currentUser } = this.props

    if (!currentUser) {
      return <Redirect to="/sign-in" noThrow />
    }

    if (!product) {
      return <Redirect to="/" noThrow />
    }

    return (
      <Layout>
        {this.renderHeader(this.props)}
        {this.renderFields(this.props)}
        {this.renderBottom(this.props)}
      </Layout>
    )
  }
}

export const ConfirmOrderScreen = withStyles(styles)(ConfirmOrderComponent)

export default connect(
  mapState,
  null,
)(props => (
  <New
    name="orders"
    render={renderProps => <ConfirmOrderScreen {...props} {...renderProps} />}
    afterCreate={handleAfterCreate(props.navigate)}
  />
))
