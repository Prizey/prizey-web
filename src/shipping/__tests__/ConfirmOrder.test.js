import React from 'react'
import renderer from 'react-test-renderer'
import ConfirmOrder, {
  ConfirmOrderComponent,
  ConfirmOrderScreen,
  handleCancel,
  handleShipIt,
  handleAfterCreate,
  mapState,
} from '../ConfirmOrder'

jest.mock('../../design/Layout/RegisterPageView')
jest.mock('design/UserBalance', () => () => null)

jest.mock('@material-ui/core/Dialog', () => ({ children, ...props }) => (
  <div {...props}>{children}</div>
))

jest.mock('react-redux', () => ({
  connect: () => Component => props => (
    <Component {...props} product={{ id: 1 }} />
  ),
}))

jest.mock('croods', () => ({
  New: props => (
    <div {...props}>
      New -
      <div>
        render -
        {props.render({
          create: jest.fn(),
          creating: false,
          error: null,
        })}
      </div>
      <div>
        renderCreated - {props.renderCreated && props.renderCreated({ id: 1 })}
      </div>
    </div>
  ),
}))

it('renders correctly', () => {
  const params = {
    classes: {},
    currentUser: {
      address: '1# street',
      city: 'porto alegre',
      clothingSize: 'M',
      fullname: 'john john',
      shoeSize: '42',
      stateProvinceRegion: 'RS',
      zipCode: '90000-000',
    },
  }

  const tree = renderer.create(<ConfirmOrder {...params} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Component correctly', () => {
  const params = {
    currentUser: {
      address: '1# street',
      city: 'porto alegre',
      clothingSize: 'M',
      fullname: 'john john',
      shoeSize: '42',
      stateProvinceRegion: 'RS',
      zipCode: '90000-000',
    },
  }

  const tree = renderer.create(<ConfirmOrderScreen {...params} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('redirects correctly', () => {
  const tree = renderer.create(<ConfirmOrderScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("redirects correctly when doesn't have product", () => {
  const tree = renderer.create(<ConfirmOrderScreen currentUser />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('check cancel action', () => {
  const navigate = jest.fn()
  handleCancel(navigate)()

  expect(navigate).toHaveBeenCalledWith('/')
})

it('check cancel action', () => {
  const navigate = jest.fn()
  handleAfterCreate(navigate)()

  expect(navigate).toHaveBeenCalledWith('/shipping-confirmation')
})

it('check shipIt action', () => {
  const create = jest.fn()
  const product = { id: 1 }

  handleShipIt(product, create)()
  expect(create).toHaveBeenCalledWith({ productId: product.id })
})

it('map the state to props', () => {
  const basket = {
    product: { image: '/mock/trump-mask.png', title: 'Trump mask' },
  }

  expect(mapState({ basket })).toEqual({ product: basket.product })
})

it('open modal when click on button', () => {
  const params = {
    classes: {},
    currentUser: true,
    product: true,
  }
  const tree = renderer.create(<ConfirmOrderComponent {...params} />).root

  tree.findByProps({ 'aria-label': 'Cancel' }).props.onClick()
  expect(tree.instance.state.dialogIsOpen).toEqual(true)
})

it('close modal when click on close', () => {
  const params = {
    classes: {},
    currentUser: true,
    product: true,
  }
  const tree = renderer.create(<ConfirmOrderComponent {...params} />).root

  tree.findByProps({ 'aria-label': 'Play again modal' }).props.close()
  expect(tree.instance.state.dialogIsOpen).toEqual(false)
})
