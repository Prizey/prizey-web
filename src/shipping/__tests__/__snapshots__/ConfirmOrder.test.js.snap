// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirects correctly 1`] = `null`;

exports[`redirects correctly when doesn't have product 1`] = `null`;

exports[`renders Component correctly 1`] = `null`;

exports[`renders correctly 1`] = `
<div
  afterCreate={[Function]}
  name="orders"
  render={[Function]}
>
  New -
  <div>
    render -
    <div
      className="Component-root-11"
    >
      <div
        className="Component-header-9"
      >
        <div
          aria-labelledby="Are you sure?"
          classes={
            Object {
              "paper": "PlayAgain-paper-22",
            }
          }
          fullWidth={true}
          onClose={[Function]}
          open={false}
        >
          <div
            className="MuiDialogTitle-root-23"
          >
            <h2
              className="MuiTypography-root-24 MuiTypography-title-30"
            >
              Are you sure?
            </h2>
          </div>
          <div
            className="MuiDialogContent-root-60"
          >
            <p
              className="MuiTypography-root-24 MuiDialogContentText-root-61 MuiTypography-subheading-31 MuiTypography-colorTextSecondary-57"
            >
              You are leaving the page and you will lose your diamonds. Are you sure?
            </p>
          </div>
          <div
            className="MuiDialogActions-root-62"
          >
            <button
              aria-label="cancel"
              autoFocus={true}
              className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-65"
              >
                Cancel
              </span>
              <span
                className="MuiTouchRipple-root-99"
              />
            </button>
            <button
              aria-label="confirm"
              className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-65"
              >
                Confirm
              </span>
              <span
                className="MuiTouchRipple-root-99"
              />
            </button>
          </div>
        </div>
        <a
          aria-label="My profile"
          className="MuiButtonBase-root-19 MuiIconButton-root-13 Component-root-12"
          href="/profile"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiIconButton-label-18"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-90 MuiSvgIcon-fontSizeLarge-98"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <g>
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33C4.62 15.49 4 13.82 4 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83z"
                />
                <path
                  d="M12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
                />
              </g>
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-99"
          />
        </a>
      </div>
      <div
        className="Component-container-6"
      >
        <div
          aria-labelledby="Are you sure?"
          classes={
            Object {
              "paper": "PlayAgain-paper-22",
            }
          }
          fullWidth={true}
          onClose={[Function]}
          open={false}
        >
          <div
            className="MuiDialogTitle-root-23"
          >
            <h2
              className="MuiTypography-root-24 MuiTypography-title-30"
            >
              Are you sure?
            </h2>
          </div>
          <div
            className="MuiDialogContent-root-60"
          >
            <p
              className="MuiTypography-root-24 MuiDialogContentText-root-61 MuiTypography-subheading-31 MuiTypography-colorTextSecondary-57"
            >
              You are leaving the page and you will lose your diamonds. Are you sure?
            </p>
          </div>
          <div
            className="MuiDialogActions-root-62"
          >
            <button
              aria-label="cancel"
              autoFocus={true}
              className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-65"
              >
                Cancel
              </span>
              <span
                className="MuiTouchRipple-root-99"
              />
            </button>
            <button
              aria-label="confirm"
              className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-65"
              >
                Confirm
              </span>
              <span
                className="MuiTouchRipple-root-99"
              />
            </button>
          </div>
        </div>
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <img
            alt="Prizey"
            className="Component-logoImage-10"
            src="/logo.png"
          />
        </a>
        <div
          className="Component-caption-4"
        />
        <div
          className="Component-children-5"
        >
          <div
            aria-labelledby="Are you sure?"
            classes={
              Object {
                "paper": "PlayAgain-paper-22",
              }
            }
            fullWidth={true}
            onClose={[Function]}
            open={false}
          >
            <div
              className="MuiDialogTitle-root-23"
            >
              <h2
                className="MuiTypography-root-24 MuiTypography-title-30"
              >
                Are you sure?
              </h2>
            </div>
            <div
              className="MuiDialogContent-root-60"
            >
              <p
                className="MuiTypography-root-24 MuiDialogContentText-root-61 MuiTypography-subheading-31 MuiTypography-colorTextSecondary-57"
              >
                This action can't be undone.
              </p>
            </div>
            <div
              className="MuiDialogActions-root-62"
            >
              <button
                aria-label="cancel"
                autoFocus={true}
                className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-65"
                >
                  Cancel
                </span>
                <span
                  className="MuiTouchRipple-root-99"
                />
              </button>
              <button
                aria-label="confirm"
                className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-65"
                >
                  Confirm
                </span>
                <span
                  className="MuiTouchRipple-root-99"
                />
              </button>
            </div>
          </div>
          <h5
            className="MuiTypography-root-24 MuiTypography-h5-40 MuiTypography-alignCenter-47 ConfirmOrderComponent-title-2"
          >
            Confirm Order
          </h5>
          <p
            className="MuiTypography-root-24 MuiTypography-body2-32 ConfirmOrderComponent-field-1"
          >
            Full Name
            :
             
            <span
              className="ConfirmOrderComponent-value-3"
            >
              john john
            </span>
          </p>
          <p
            className="MuiTypography-root-24 MuiTypography-body2-32 ConfirmOrderComponent-field-1"
          >
            Address
            :
             
            <span
              className="ConfirmOrderComponent-value-3"
            >
              1# street
            </span>
          </p>
          <p
            className="MuiTypography-root-24 MuiTypography-body2-32 ConfirmOrderComponent-field-1"
          >
            City
            :
             
            <span
              className="ConfirmOrderComponent-value-3"
            >
              porto alegre
            </span>
          </p>
          <p
            className="MuiTypography-root-24 MuiTypography-body2-32 ConfirmOrderComponent-field-1"
          >
            State / Province / Region
            :
             
            <span
              className="ConfirmOrderComponent-value-3"
            >
              RS
            </span>
          </p>
          <p
            className="MuiTypography-root-24 MuiTypography-body2-32 ConfirmOrderComponent-field-1"
          >
            Postal Code / ZIP
            :
             
            <span
              className="ConfirmOrderComponent-value-3"
            />
          </p>
          <p
            className="MuiTypography-root-24 MuiTypography-body2-32 ConfirmOrderComponent-field-1"
          >
            Clothing Size
            :
             
            <span
              className="ConfirmOrderComponent-value-3"
            >
              M
            </span>
          </p>
          <p
            className="MuiTypography-root-24 MuiTypography-body2-32 ConfirmOrderComponent-field-1"
          >
            Shoe Size
            :
             
            <span
              className="ConfirmOrderComponent-value-3"
            >
              42
            </span>
          </p>
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          />
          <button
            className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-contained-75 MuiButton-containedPrimary-76 MuiButton-raised-78 MuiButton-raisedPrimary-79 MuiButton-fullWidth-89"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-65"
            >
              SHIP IT!
            </span>
            <span
              className="MuiTouchRipple-root-99"
            />
          </button>
          <button
            aria-label="Cancel"
            className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-outlined-72 MuiButton-outlinedPrimary-73 MuiButton-fullWidth-89"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-65"
            >
              CANCEL
            </span>
            <span
              className="MuiTouchRipple-root-99"
            />
          </button>
        </div>
        <div
          className="Component-footer-7"
        >
          <div
            aria-labelledby="Are you sure?"
            classes={
              Object {
                "paper": "PlayAgain-paper-22",
              }
            }
            fullWidth={true}
            onClose={[Function]}
            open={false}
          >
            <div
              className="MuiDialogTitle-root-23"
            >
              <h2
                className="MuiTypography-root-24 MuiTypography-title-30"
              >
                Are you sure?
              </h2>
            </div>
            <div
              className="MuiDialogContent-root-60"
            >
              <p
                className="MuiTypography-root-24 MuiDialogContentText-root-61 MuiTypography-subheading-31 MuiTypography-colorTextSecondary-57"
              >
                You are leaving the page and you will lose your diamonds. Are you sure?
              </p>
            </div>
            <div
              className="MuiDialogActions-root-62"
            >
              <button
                aria-label="cancel"
                autoFocus={true}
                className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-65"
                >
                  Cancel
                </span>
                <span
                  className="MuiTouchRipple-root-99"
                />
              </button>
              <button
                aria-label="confirm"
                className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-65"
                >
                  Confirm
                </span>
                <span
                  className="MuiTouchRipple-root-99"
                />
              </button>
            </div>
          </div>
          <a
            className="Component-footerLink-8"
            href="/terms_of_service"
            onClick={[Function]}
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-24 MuiTypography-caption-34 MuiTypography-colorTextSecondary-57 MuiTypography-alignCenter-47"
            >
              TERMS
            </span>
          </a>
          <div
            aria-labelledby="Are you sure?"
            classes={
              Object {
                "paper": "PlayAgain-paper-22",
              }
            }
            fullWidth={true}
            onClose={[Function]}
            open={false}
          >
            <div
              className="MuiDialogTitle-root-23"
            >
              <h2
                className="MuiTypography-root-24 MuiTypography-title-30"
              >
                Are you sure?
              </h2>
            </div>
            <div
              className="MuiDialogContent-root-60"
            >
              <p
                className="MuiTypography-root-24 MuiDialogContentText-root-61 MuiTypography-subheading-31 MuiTypography-colorTextSecondary-57"
              >
                You are leaving the page and you will lose your diamonds. Are you sure?
              </p>
            </div>
            <div
              className="MuiDialogActions-root-62"
            >
              <button
                aria-label="cancel"
                autoFocus={true}
                className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-65"
                >
                  Cancel
                </span>
                <span
                  className="MuiTouchRipple-root-99"
                />
              </button>
              <button
                aria-label="confirm"
                className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-65"
                >
                  Confirm
                </span>
                <span
                  className="MuiTouchRipple-root-99"
                />
              </button>
            </div>
          </div>
          <a
            className="Component-footerLink-8"
            href="https://app.termly.io/document/privacy-policy/8192f107-a2dc-4548-bf83-97907c6d83c3"
            onClick={[Function]}
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-24 MuiTypography-caption-34 MuiTypography-colorTextSecondary-57 MuiTypography-alignCenter-47"
            >
              PRIVACY
            </span>
          </a>
          <div
            aria-labelledby="Are you sure?"
            classes={
              Object {
                "paper": "PlayAgain-paper-22",
              }
            }
            fullWidth={true}
            onClose={[Function]}
            open={false}
          >
            <div
              className="MuiDialogTitle-root-23"
            >
              <h2
                className="MuiTypography-root-24 MuiTypography-title-30"
              >
                Are you sure?
              </h2>
            </div>
            <div
              className="MuiDialogContent-root-60"
            >
              <p
                className="MuiTypography-root-24 MuiDialogContentText-root-61 MuiTypography-subheading-31 MuiTypography-colorTextSecondary-57"
              >
                You are leaving the page and you will lose your diamonds. Are you sure?
              </p>
            </div>
            <div
              className="MuiDialogActions-root-62"
            >
              <button
                aria-label="cancel"
                autoFocus={true}
                className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-65"
                >
                  Cancel
                </span>
                <span
                  className="MuiTouchRipple-root-99"
                />
              </button>
              <button
                aria-label="confirm"
                className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-65"
                >
                  Confirm
                </span>
                <span
                  className="MuiTouchRipple-root-99"
                />
              </button>
            </div>
          </div>
          <a
            className="Component-footerLink-8"
            href="/faq"
            onClick={[Function]}
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-24 MuiTypography-caption-34 MuiTypography-colorTextSecondary-57 MuiTypography-alignCenter-47"
            >
              FAIRNESS / FAQ
            </span>
          </a>
          <div
            aria-labelledby="Are you sure?"
            classes={
              Object {
                "paper": "PlayAgain-paper-22",
              }
            }
            fullWidth={true}
            onClose={[Function]}
            open={false}
          >
            <div
              className="MuiDialogTitle-root-23"
            >
              <h2
                className="MuiTypography-root-24 MuiTypography-title-30"
              >
                Are you sure?
              </h2>
            </div>
            <div
              className="MuiDialogContent-root-60"
            >
              <p
                className="MuiTypography-root-24 MuiDialogContentText-root-61 MuiTypography-subheading-31 MuiTypography-colorTextSecondary-57"
              >
                You are leaving the page and you will lose your diamonds. Are you sure?
              </p>
            </div>
            <div
              className="MuiDialogActions-root-62"
            >
              <button
                aria-label="cancel"
                autoFocus={true}
                className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-65"
                >
                  Cancel
                </span>
                <span
                  className="MuiTouchRipple-root-99"
                />
              </button>
              <button
                aria-label="confirm"
                className="MuiButtonBase-root-19 MuiButton-root-64 MuiButton-text-66 MuiButton-textPrimary-67 MuiButton-flat-69 MuiButton-flatPrimary-70 MuiDialogActions-action-63"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiButton-label-65"
                >
                  Confirm
                </span>
                <span
                  className="MuiTouchRipple-root-99"
                />
              </button>
            </div>
          </div>
          <a
            className="Component-footerLink-8"
            href="/all-prizes"
            onClick={[Function]}
            rel="noopener noreferer"
            target="_blank"
          >
            <span
              className="MuiTypography-root-24 MuiTypography-caption-34 MuiTypography-colorTextSecondary-57 MuiTypography-alignCenter-47"
            >
              ALL PRIZES
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div>
    renderCreated - 
  </div>
</div>
`;
